integration-test/src/test/scala/io/phdata/clients/RoleClientIntegrationSpec.scala

```diff
@@ -0,0 +1,210 @@
+package io.phdata.clients
+
+import java.util.UUID
+
+import cats.effect.IO
+import io.phdata.itest.fixtures.{HiveTest, IntegrationTest, _}
+import io.phdata.models.{Manager, ReadOnly, ReadWrite}
+import io.phdata.services.UGILoginContextProvider
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+import cats.implicits._
+
+class RoleClientIntegrationSpec extends FlatSpec with Matchers with HiveTest with IntegrationTest with BeforeAndAfterAll{
+
+  val hiveClient = new HiveClientImpl[IO](new UGILoginContextProvider(itestConfig), hiveTransactor)
+  val roleClient = new RoleClientImpl[IO](hiveTransactor, null, new UGILoginContextProvider(itestConfig))
+
+  val role = s"test_role_${UUID.randomUUID().toString.take(8)}"
+  val group = "integration_test_group"
+  val testDatabase = "sentry_client_integration_test"
+  val location = "integration_test"
+
+  override def beforeAll() {
+    val init = for {
+      existingDBs <- hiveClient.showDatabases()
+      _ <- if (!existingDBs.contains(testDatabase)) {
+        hiveClient.createDatabase(testDatabase, "/tmp", "a comment", Map("pii_data" -> "true", "pci_data" -> "false"))
+      } else {
+        ().pure[IO]
+      }
+    } yield ()
+
+    init.unsafeRunSync()
+  }
+
+  override def afterAll() {
+    val cleanUp = for {
+     roles <- roleClient.roles
+     _ <- roles.filter(_.startsWith("test_role_")).map(testRole => roleClient.dropRole(testRole)).pure[IO]
+    } yield ()
+
+    cleanUp.unsafeRunSync()
+  }
+
+  behavior of "Sentry Client"
+
+  it should "list roles" in {
+    val testRoles = List(
+      s"test_role_${UUID.randomUUID().toString.take(8)}",
+      s"test_role_${UUID.randomUUID().toString.take(8)}",
+      s"test_role_${UUID.randomUUID().toString.take(8)}"
+    )
+
+    testRoles.foreach(role =>
+      roleClient.createRole(role).unsafeRunSync()
+    )
+
+    val result = roleClient.roles.unsafeRunSync()
+
+    result should not be empty
+    testRoles.foreach(role =>
+      result should contain(role)
+    )
+
+    testRoles.foreach(role =>
```

Comment:
This should either go in an `after` method or in a `finally` block so everything gets cleaned up, even on exception

Reply:
This just validates the role. This block starting on line 35 cleans them up
```
  override def afterAll() {
    val cleanUp = for {
     roles <- roleClient.roles
     _ <- roles.filter(_.startsWith("arch_test_role_")).map(testRole => roleClient.dropRole(testRole)).pure[IO]
    } yield ()

    cleanUp.unsafeRunSync()
  }
```